File: Ejercicio3.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    vector<int> arrayNumeros;
    for (int i = 1; i <= 10; i++) arrayNumeros.push_back(i);

    cout << "Recorrido inicial:" << endl;
    for (int i = 0; i < arrayNumeros.size(); i++) {
        cout << "[" << i << "]: " << arrayNumeros[i] << endl;
    }

    cout << "Buscando el valor 8:" << endl;
    for (int i = 0; i < arrayNumeros.size(); i++) {
        if (arrayNumeros[i] == 8) {
            cout << "¡Encontrado! El valor 8 está en la posición " << i << endl;
            break;
        }
    }

    arrayNumeros.insert(arrayNumeros.begin() + 5, 22);

    cout << "Recorrido después de la inserción:" << endl;
    for (int i = 0; i < arrayNumeros.size(); i++) {
        cout << "[" << i << "]: " << arrayNumeros[i] << endl;
    }

    cout << "Recorrido inverso:" << endl;
    for (int i = arrayNumeros.size() - 1; i >= 0; i--) {
        cout << "[" << i << "]: " << arrayNumeros[i] << endl;
    }

    return 0;
}

File: Ejercicio3.py
array_numeros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

print("Recorrido inicial:")
for i, numero in enumerate(array_numeros):
    print(f"[{i}]: {numero}")

print("Buscando el valor 8:")
for i, numero in enumerate(array_numeros):
    if numero == 8:
        print(f"¡Encontrado! El valor 8 está en la posición {i}")
        break

array_numeros.insert(5, 22)

print("Recorrido después de la inserción:")
for i, numero in enumerate(array_numeros):
    print(f"[{i}]: {numero}")

print("Recorrido inverso:")
for i, numero in enumerate(reversed(array_numeros)):
    print(f"[{len(array_numeros)-1-i}]: {numero}")